### ISSUE TITLE
Windows XP support?

### ISSUE THREAD

**Issue by zturtleman** [libsdl-org/SDL#8666]
re: Fixed building with older Windows SDK https://github.com/libsdl-org/SDL/commit/f00ecf5f190feb6db0727b0c93f7a609f70be76d

These flags require Vista. I'm mentioning it in case SDL 3 is intended to work on XP.

https://github.com/libsdl-org/SDL/blob/f00ecf5f190feb6db0727b0c93f7a609f70be76d/src/video/windows/SDL_windowsevents.c#L176

[MapVirtualKeyA documentation](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-mapvirtualkeya)

https://github.com/libsdl-org/SDL/blob/f00ecf5f190feb6db0727b0c93f7a609f70be76d/src/video/windows/SDL_windowsevents.c#L878

[WideCharToMultiByte documentation](https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-widechartomultibyte). If I remember correctly XP always returns an error with the flag set.

**Comment by icculus** [comment]
For small fixes, we don’t mind keeping XP support, but I don’t think heroic efforts will be made.

Not sure where a lack of UTF-8 support in WideCharToMultiByte falls on that scale, though.

**Comment by RamonUnch** [comment]
> Not sure where a lack of UTF-8 support in WideCharToMultiByte falls on that scale, though.

Windows XP handles UTF-8 just fine (even older windows handled it), it will simply fail if the `WC_ERR_INVALID_CHARS` flag is used (i just tested to be sure). This flag could look useless to convert from UTF-16 to UTF-8 because any UTF-16 string can be converted to UTF-8. However this flag allows failure on invalid UTF-16 strings, I tested with unmatched surrogate pair, so it does play a role here.

Maybe if someone is using an old SDK he might be specifically targeting WinXP specifically, hence defining to 0 these macros might make more sense, but the user might as well do it himself.

**Comment by icculus** [comment]
A good example of continuing XP support would be: if we see we're running XP at runtime, and we drop that flag, and you're on your own if you hit a weird corner case with invalid UTF-16 values.

Doesn't take much code, and unblocks legacy support? Let's do that.

If other unrelated things make XP support untenable, let's not change this code, but if _that's_ the only blocker, it's an easy fix.

**Comment by icculus** [comment]
Likewise for `MAPVK_VK_TO_VSC_EX`, we can probably have a `AreWeRunningOnWinXP` flag that makes it use `MAPVK_VK_TO_VSC` instead, and good luck on the weird corner cases.

So let's decide if there's any more major code that eliminates XP support, and if _that's it_, let's patch that in and close this bug. If there's another actual runtime showstopper, let's not bother and just leave this on WONTFIX.

I don't know the answer to this, someone else will have to say.

Note that SDL2 still supports XP, so it's possible this is literally _everything_ that broke XP support in SDL3.

**Comment by icculus** [comment]
Okay, check #8929, everyone. If it's reasonable and resolves all WinXP issues, let's merge it, otherwise let's close this issue as WONTFIX.
